### PR TITLE
NIFI-5589 : Clarify PutMongo documentation

### PR DESCRIPTION
**Pull request by VijetaH**
Thank you for submitting a contribution to Apache NiFi.

In order to streamline the review of the contribution we ask you
to ensure the following steps have been taken:

### For all changes:
- [x] Is there a JIRA ticket associated with this PR? Is it referenced 
     in the commit message?

- [x] Does your PR title start with NIFI-XXXX where XXXX is the JIRA number you are trying to resolve? Pay particular attention to the hyphen "-" character.

- [x] Has your PR been rebased against the latest commit within the target branch (typically master)?

- [ ] Is your initial contribution a single, squashed commit?

### For code changes:
- [ ] Have you ensured that the full suite of tests is executed via mvn -Pcontrib-check clean install at the root nifi folder?
- [ ] Have you written or updated unit tests to verify your changes?
- [ ] If adding new dependencies to the code, are these dependencies licensed in a way that is compatible for inclusion under [ASF 2.0](http://www.apache.org/legal/resolved.html#category-a)? 
- [ ] If applicable, have you updated the LICENSE file, including the main LICENSE file under nifi-assembly?
- [ ] If applicable, have you updated the NOTICE file, including the main NOTICE file found under nifi-assembly?
- [ ] If adding new Properties, have you added .displayName in addition to .name (programmatic access) for each of the new properties?

### For documentation related changes:
- [ ] Have you ensured that format looks appropriate for the output in which it is rendered?

### Note:
Please ensure that once the PR is submitted, you check travis-ci for build issues and submit an update to your PR as soon as possible.

### CODE REVIEW

**Comment by josephxsxn** [comment]
Thanks for submitting this documentation change. Can you please squash all 3 of your commits into 1?

https://github.com/todotxt/todo.txt-android/wiki/Squash-All-Commits-Related-to-a-Single-Issue-into-a-Single-Commit

**Comment by alopresto** [comment]
Hi @josephxsxn I don't think we have to request that any more as GitHub shows the consolidated diff in one view, and rebasing & force-pushing to a branch that is used for a PR destroys history and can mess up the reviewer comments. The committer rebases & squashes the commits when they merge them.

**Comment by VijetaH** [comment]
@alopresto Can you please review the changes?

**Comment by alopresto** [comment]
I looked at the code, and the documentation change is syntactically valid, but I am not familiar enough with MongoDB to assert that the content is correct. You will need a reviewer who is a MongoDB user. There are a few in the community and they look for PRs with _Mongo_ in the title quite regularly.

**Comment by VijetaH** [comment]
@MikeThomsen Could you please review MongoDB document changes?

File: nifi-nar-bundles/nifi-mongodb-bundle/nifi-mongodb-processors/src/main/java/org/apache/nifi/processors/mongodb/PutMongo.java [hunk 1]
```diff
@@ -112,8 +112,10 @@
         .allowableValues(UPDATE_WITH_DOC, UPDATE_WITH_OPERATORS)
         .defaultValue(UPDATE_WITH_DOC.getValue())
         .description("Choose an update mode. You can either supply a JSON document to use as a direct replacement " +
-                "or specify a document that contains update operators like $set and $unset")
-        .build();
+                "or specify a document that contains update operators like $set, $unset, and $inc."+
+                        "When Operators mode is enabled, the flowfile content is expected to be the operator part"+
+                        "for example: {$set:{\"key\": \"value\"},$inc:{\"count\":1234}} and Update query has to be the record to be updated")
```
Comment (MikeThomsen) [inline nifi-nar-bundles/nifi-mongodb-bundle/nifi-mongodb-processors/src/main/java/org/apache/nifi/processors/mongodb/PutMongo.java]: The part here after `and` is somewhat ambiguous and could potentially confuse people doing upserts. Other than that, it looks fine.

Comment (MikeThomsen) [reply nifi-nar-bundles/nifi-mongodb-bundle/nifi-mongodb-processors/src/main/java/org/apache/nifi/processors/mongodb/PutMongo.java]: I'd suggest we strike that part and replace it with something like this:

> and the update query will come from the configured `Update Query` property.

If that works for you, I can make the change myself.

Comment (VijetaH) [reply nifi-nar-bundles/nifi-mongodb-bundle/nifi-mongodb-processors/src/main/java/org/apache/nifi/processors/mongodb/PutMongo.java]: @MikeThomsen yes please go ahead with the changes.

Comment (MikeThomsen) [reply nifi-nar-bundles/nifi-mongodb-bundle/nifi-mongodb-processors/src/main/java/org/apache/nifi/processors/mongodb/PutMongo.java]: Can you git cherry-pick this onto a branch named NIFI-5589? It's currently sitting on your master.

Comment (MikeThomsen) [reply nifi-nar-bundles/nifi-mongodb-bundle/nifi-mongodb-processors/src/main/java/org/apache/nifi/processors/mongodb/PutMongo.java]: (Just push that branch to GitHub, I'll fetch it and merge)

**Comment by MikeThomsen** [comment]
Merged. Thanks for the contribution.

**Pull request closed without merging**